packages/lib/src/components/internal/Address/Address.tsx

```diff
@@ -30,17 +31,49 @@ export default function Address(props: AddressProps) {
 
     const requiredFieldsSchema = specifications.getAddressSchemaForCountryFlat(props.countryCode).filter(field => requiredFields.includes(field));
 
-    const { data, errors, valid, isValid, handleChangeFor, triggerValidation } = useForm<AddressData>({
+    const [hasSelectedAddress, setHasSelectedAddress] = useState(false);
+
+    const [useManualAddress, setUseManualAddress] = useState(false);
+
+    const [searchErrorMessage, setSearchErrorMessage] = useState('');
+
+    const showAddressSearch = !!props.onAddressLookup;
+
+    const showAddressFields = props.onAddressLookup ? hasSelectedAddress || useManualAddress : true;
+
+    const { data, errors, valid, isValid, handleChangeFor, triggerValidation, setData } = useForm<AddressData>({
         schema: requiredFieldsSchema,
         defaultData: props.data,
         // Ensure any passed validation rules are merged with the default ones
         rules: { ...getAddressValidationRules(specifications), ...props.validationRules },
         formatters: addressFormatters
     });
 
+    const setSearchData = selectedAddress => {
+        const propsKeysToProcess = ADDRESS_SCHEMA;
+        propsKeysToProcess.forEach(propKey => {
+            // Make sure the data provided by the merchant is always strings
+            const providedValue = selectedAddress[propKey];
+            if (providedValue === null || providedValue === undefined) return;
+            // Cast everything to string
+            setData(propKey, String(providedValue));
+            triggerValidation();
+        });
+        setHasSelectedAddress(true);
+    };
+
+    const onManualAddress = () => {
+        setUseManualAddress(true);
+    };
```

Comment:
You can wrap those functions in `useCallback` . They can/will end up being recreated multiple times without need for that